### PR TITLE
Fix `make docs`

### PR DESCRIPTION
**Pull request by mirekdlugosz**
Travis failed in last couple of PRs on `make docs` step. I don't know why it complains now, it's not like I'm reverting any recent changes - these things were here for years already.

There are two important changes.

One, there are 2 new steps in travis build process. In total, they add around 10 seconds to travis run, which I consider acceptable. But that slowdown is here. And I'm not 100% sure if that's the correct solution to the problem, but I couldn't find anything better.

Two, I removed tests/foreman/ui/ from generated docs. The problem is that some files use global module code to tell pytest to skip entire module when condition is met (e.g. [test_computeresource_ec2](https://github.com/SatelliteQE/robottelo/blob/82dce81bbf6fcbd98c77086efeb0d57b07b2595d/tests/foreman/ui/test_computeresource_ec2.py#L32)). When sphinx executes that, it considers `pytest.skip()` to raise exception and fails entire build on that.

I did not find a way to ignore this exception. Marking specific tests with `@pytest.mark.skipif` does work, but is awkward for files like `test_contenthost.py` (which has 24 tests). I **could** modify `scripts/gen_api_docs.sh` to ignore these specific files with that pattern, but decided to throw baby out with bathwater and just skip the whole thing.

The real solution is already implemented in #7656 , which removed `sphinx-autodoc` and it's insane idea of importing modules.

------------------

Changes in `docs/api` are mostly auto-generated by `scripts/gen_api_docs.sh`.

```bash
$ make docs
make[1]: Entering directory '/home/mzalewsk/sources/robottelo/docs'
cd ..; ./scripts/gen_api_docs.sh
sphinx-build -W -b html -d _build/doctrees   . _build/html
Running Sphinx v2.4.1
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 27 source files that are out of date
updating environment: [new config] 27 added, 0 changed, 0 removed
reading sources... [ 62%] api/tests.foreman.ui                                                                                                                                                 
Build finished. The HTML pages are in _build/html.
make[1]: Leaving directory '/home/mzalewsk/sources/robottelo/docs'
```
(But the real evidence is in travis build status)

### CODE REVIEW

**Comment by mirekdlugosz** [comment]
@SatelliteQE/quality-engineers anyone has idea how to fix travis problem here?
```
$ make docs
...
Warning, treated as error:
autodoc: failed to import module 'test_provisioning_computeresource' from module 'tests.foreman.longrun'; the following exception was raised:
pycurl: libcurl link-time ssl backend (gnutls) is different from compile-time ssl backend (openssl)
Makefile:49: recipe for target 'html' failed
```

I tried adding one these two lines to `.travis.yaml`, but no luck:
```
env:
    - PYCURL_SSL_LIBRARY=gnutls
```
```
env:
    - PYCURL_SSL_LIBRARY=openssl
```

**Comment by vijay8451** [comment]
@mirekdlugosz plz give a try to below:
```
sudo yum reinstall gnutls
pip uninstall pycurl
export PYCURL_SSL_LIBRARY=gnutls
pip install --compile --no-cache-dir pycurl
```

**Comment by mirekdlugosz** [comment]
@vijay8451 that can be done in Travis, too. I actually did exactly that in newest revision of PR. I'm not sure if that's the best way, but it does fulfill the immediate need.

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+    - PYCURL_SSL_LIBRARY=gnutls
 python:
     - "3.6"
+before_install:
+    - sudo apt-get install -y libgnutls-dev
 install:
     - pip install -U pip
     - pip install -r requirements.txt codecov flake8 sphinx tox-travis manage pre-commit
+    - pip uninstall -y pycurl
+    - pip install --compile --no-cache-dir pycurl
```
Comment (vijay8451) [inline .travis.yml]:
```suggestion
    - pip install -U pip 
    - pip uninstall -y pycurl
    - pip install --compile --no-cache-dir pycurl
    - pip install -r requirements.txt codecov flake8 sphinx tox-travis manage pre-commit 
```

I think there should not be any harm to add these steps (unless anything more related to pycurl compile/recompile) ..
However could you please set the in above order and try once more to see if all works fine ?

Comment (mirekdlugosz) [reply .travis.yml]: It makes no difference for travis.
It also makes no difference for me. It can be this way, still works.

Comment (vijay8451) [reply .travis.yml]: the idea was to see if compile works perfectly in different order too  before install the `pip install -r requirements.txt codecov flake8 sphinx tox-travis manage pre-commit `

as travis is looks satisfied hence LGTM :+1:

**Pull request merged**